src/Bridge/Symfony/Bundle/Test/Response.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\Test;
+
+use ApiPlatform\Core\Bridge\Symfony\Bundle\Test\Exception\ClientException;
+use Symfony\Component\BrowserKit\Response as BrowserKitResponse;
+use Symfony\Component\HttpClient\Exception\JsonException;
+use Symfony\Component\HttpFoundation\Response as HttpFoundationResponse;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * HTTP Response.
+ *
+ * @internal
+ *
+ * Partially copied from \Symfony\Component\HttpClient\Response\ResponseTrait
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class Response implements ResponseInterface
+{
+    private $httpFoundationResponse;
+    private $browserKitResponse;
+    private $info;
+    private $content;
+    private $jsonData;
+
+    public function __construct(HttpFoundationResponse $httpFoundationResponse, BrowserKitResponse $browserKitResponse, array $info)
+    {
+        $this->httpFoundationResponse = $httpFoundationResponse;
+        $this->browserKitResponse = $browserKitResponse;
+
+        $this->headers = $httpFoundationResponse->headers->all();
+        $this->content = $httpFoundationResponse->getContent();
+        $this->info = $info + [
+            'http_code' => $httpFoundationResponse->getStatusCode(),
+            'error' => null,
+            'raw_headers' => $this->headers,
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getInfo(string $type = null)
+    {
+        if ($type) {
+            return $this->info[$type] ?? null;
+        }
+
+        return $this->info;
+    }
+
+    /**
+     * Checks the status, and try to extract message if appropriate.
+     */
+    private function checkStatusCode()
+    {
+        $message = 'An error '.$this->info['http_code'].' occured.';
+        if (isset($this->headers['content-type'][0]) && false !== preg_match('#^application/(?:.+\+)?json#', $this->headers['content-type'][0])) {
+            if ($json = json_decode($this->content, true)) {
+                // Try to extract the error message from Hydra or RFC 7807 error structures
+                $message = $json['hydra:description'] ?? $json['hydra:title'] ?? $json['detail'] ?? $json['title'] ?? $message;
```

Comment:
:disappointed: I think we should only extract the correct error message based on the `Content-Type`.

Reply:
It will add extra complexity for no benefits. If one of these key exist, and a faulty HTTP status code is set (the case here), this key will definitely contain an error, regardless of the MIME type.